project/future/PLAN-auth-future-improvements.md

```
@@ -0,0 +1,463 @@
+- Student authentication via Edu-ID
+- Automatic participant account creation and linking
+- Session isolation between lecturer and student contexts
+- Context switching with proper session cleanup
+- Assessment mode detection and appropriate UI display
+
+---
+
+## Phase 1: Security Hardening & Validation
+
+### 1.1 Enhanced URL Validation
+**Priority: HIGH**
```

markdownlint-cli2 (0.17.2)

Emphasis used instead of a heading

(MD036, no-emphasis-as-heading)